Add support for using a KVM storage pool instead of storing as a file.

jcodybaker opened the pull request:
This patch optionally uses libvirt's storage pools and volumes for the root volume.  This provides a standardized way to use ZFS, LVM and other libvirt pool drivers for storage as opposed to directly writing QEMU COW files.  Beyond the direct benefits, this may useful if remote libvirt's machines are ever supported (because direct file manipulation isn't required).  

This is the first I've worked with GO so please feel free to correct any style or substance.

dhiltgen commented on kvm.go:
```diff
@@ -684,6 +725,44 @@ func createDiskImage(dest string, size int, r io.Reader) error {
+		return err
+	}
+	xml := fmt.Sprintf(newVolumeXML, d.StorageVolumeName, sizeBytes, sizeBytes)
+	volume, err := pool.StorageVolCreateXML(xml,0);
```
I believe you'll need a corresponding call to `StorageVolDelete(...)` http://libvirt.org/html/libvirt-libvirt-storage.html#virStorageVolDelete most likely from within `Remove()` - In my testing of this PR it looks like it's leaking volumes. Upon removal of the machine, the storage volume still exists in the pool.

mikefaille commented:
@jcodybaker Do you have some news ?

jcodybaker commented:
Sorry for the delay. I've updated this pull request to address the volume cleanup on "docker-machine rm".  I've also updated the PR to use the updated libvirt-go library, and to include better documentation.

yan-zhuang commented:
May I ask the status of this PR? It seems not being merged.

jcodybaker commented:
@yan-zhuang  - The PR was never merged.  There was a request for changes, the changes were made, but no followup review was made. This PR is from 2016, and it looks like the last commit on master was in 2017. I don't think it's likely this will be merged or is useful to most folks.  If it's useful to you you're welcome to me reopen the PR, or to fork the repo w/ the PR.

yan-zhuang commented:
@jcodybaker - Thanks for replying back. Yes indeed, the PR introduced a very useful feature, and I saw an [issue ](https://github.com/dhiltgen/docker-machine-kvm/issues/70) asking about it. It would be nice if the feature could be added into the main repo.
